<?xml version="1.0" encoding="UTF-8"?>
<!-- This does not replace URDF, and is not an extension of URDF.
     This is a format for representing semantic information about the robot structure.
     A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined.
-->
<robot name="dual_arm_robot">

    <!-- GROUPS: Define groups for each arm -->
    <group name="arm_a_group">
        <joint name="arm_a_world2base"/>
        <joint name="arm_a_arm_joint_0"/>
        <joint name="arm_a_arm_joint_1"/>
        <joint name="arm_a_arm_joint_2"/>
        <joint name="arm_a_arm_joint_3"/>
        <joint name="arm_a_arm_joint_4"/>
        <joint name="arm_a_arm_joint_5"/>
    </group>

    <group name="arm_b_group">
        <joint name="arm_b_world2base"/>
        <joint name="arm_b_arm_joint_0"/>
        <joint name="arm_b_arm_joint_1"/>
        <joint name="arm_b_arm_joint_2"/>
        <joint name="arm_b_arm_joint_3"/>
        <joint name="arm_b_arm_joint_4"/>
        <joint name="arm_b_arm_joint_5"/>
    </group>

    <!-- END EFFECTORS: Define end effectors for each arm -->
    <end_effector name="arm_a_end_effector" parent_link="arm_a_link5" group="arm_a_group"/>
    <end_effector name="arm_b_end_effector" parent_link="arm_b_link5" group="arm_b_group"/>

    <!-- DISABLE COLLISIONS: Define adjacent links and links that should never collide -->
    <!-- arm_a -->
    <disable_collisions link1="arm_a_base_link" link2="arm_a_link1" reason="Adjacent"/>
    <disable_collisions link1="arm_a_link1" link2="arm_a_link2" reason="Adjacent"/>
    <disable_collisions link1="arm_a_link1" link2="arm_a_link4" reason="Never"/>
    <disable_collisions link1="arm_a_link2" link2="arm_a_link3" reason="Adjacent"/>
    <disable_collisions link1="arm_a_link3" link2="arm_a_link4" reason="Adjacent"/>
    <disable_collisions link1="arm_a_link4" link2="arm_a_link5" reason="Adjacent"/>
    <disable_collisions link1="arm_a_link4" link2="arm_a_link6" reason="Never"/>
    <disable_collisions link1="arm_a_link5" link2="arm_a_link6" reason="Adjacent"/>

    <!-- arm_b -->
    <disable_collisions link1="arm_b_base_link" link2="arm_b_link1" reason="Adjacent"/>
    <disable_collisions link1="arm_b_link1" link2="arm_b_link2" reason="Adjacent"/>
    <disable_collisions link1="arm_b_link1" link2="arm_b_link4" reason="Never"/>
    <disable_collisions link1="arm_b_link2" link2="arm_b_link3" reason="Adjacent"/>
    <disable_collisions link1="arm_b_link3" link2="arm_b_link4" reason="Adjacent"/>
    <disable_collisions link1="arm_b_link4" link2="arm_b_link5" reason="Adjacent"/>
    <disable_collisions link1="arm_b_link4" link2="arm_b_link6" reason="Never"/>
    <disable_collisions link1="arm_b_link5" link2="arm_b_link6" reason="Adjacent"/>

</robot>
